app/src/test/java/gateway/ITShareManagement.java

```diff
@@ -0,0 +1,144 @@
+import gateway.soap.request.Authorization;
+import gateway.soap.request.ReqFile;
+import gateway.soap.request.ReqShareFile;
+import gateway.soap.response.ResStatus;
+import gateway.testutils.TestUtilAuth;
+import gateway.testutils.TestUtilConfig;
+import gateway.testutils.TestUtilShare;
+import java.util.UUID;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class ITShareManagement
+{
+
+	@BeforeEach void setup () { Config.initializeFromEnv (); }
+
+	@Test void ShareFile ()
+	{
+
+		String username1 = UUID.randomUUID ().toString ();
+		String username2 = UUID.randomUUID ().toString ();
+
+		// register
+		String tokenUser1 = TestUtilAuth.registerAndLoginUserSuccess (username1);
+		String tokenUser2 = TestUtilAuth.registerAndLoginUserSuccess (username2);
+
+		// create a file and aux file
+		ResSaveFile resSaveFile1 = TestUtilAuth.createFile (tokenUser1);
+		ResSaveFile resSaveFile2 = TestUtilAuth.createFile (tokenUser2);
+
+		// 204
+		ReqShareFile reqShareFile =
+			TestUtilShare.createShareFile (resSaveFile1, username2, tokenUser1);
+		ResStatus res = CtrlShareFile.share_file (reqShareFile);
+		assertEquals (204, res.code, "The file have been shared");
+
+		// 409
+		res = CtrlShareFile.share_file (reqShareFile);
+		assertEquals (409, res.code, "The file is already shared with the given user.");
+
+		// 403
+		reqShareFile.fileUUID = resSaveFile2.fileUUID;
+		res = CtrlShareFile.share_file (reqShareFile);
+		assertEquals (403, res.code, "The file is not owned by the user.");
+
+		// 401
+		reqShareFile.token = "token_invalid";
+		res = CtrlShareFile.share_file (reqShareFile);
+		assertEquals (401, res.code, "unauthorized");
+
+		// 400
+		reqShareFile.otherUsername = null;
+		res = CtrlShareFile.share_file (reqShareFile);
+		assertEquals (
+			400, res.code,
+			"The owner_uuid or file_uuid were not a valid UUID or the JSON body does't fullfill the validations.");
+	}
+
+	@Test void shareList ()
+	{
+		String username1 = UUID.randomUUID ().toString ();
+		String username2 = UUID.randomUUID ().toString ();
+
+		// Register
+		String tokenUser1 = TestUtilAuth.registerAndLoginUserSuccess (username1);
+		Authorization authorization = new Authorization (tokenUser1);
+		TestUtilAuth.registerAndLoginUserSuccess (username2);
+
+		// Create a file
+		ResSaveFile resSaveFile1 = TestUtilAuth.createFile (tokenUser1);
+
+		// Share file
+		ReqShareFile reqShareFile =
+			TestUtilShare.createShareFile (resSaveFile1, username2, tokenUser1);
+		ResStatus res = CtrlShareFile.share_file (reqShareFile);
+		assertEquals (204, res.code, "The file have been shared");
+
+		// List of shared with users successfully
+		assertEquals (
+			200, CtrlShareList.share_list (authorization).code, "Ok. The directory was listed.");
+
+		TestUtilConfig.makeInvalidMetadata ();
+		assertEquals (500, CtrlShareList.share_list (authorization).code, "Can't reach metadata");
+	}
+
+	@Test void shareListWhitWho ()
```

Comment:
Somehow this typo passed the previous revision, please rename it to `shareListWithWho`